sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/JacksonHttpSessionOAuth2AuthorizedClientRepository.java

```diff
@@ -77,6 +78,6 @@ private Map<String, OAuth2AuthorizedClient> getAuthorizedClients(HttpServletRequ
                        .map(s -> s.getAttribute(AUTHORIZED_CLIENTS_ATTR_NAME))
                        .map(Object::toString)
                        .map(SerializerUtils::deserializeOAuth2AuthorizedClientMap)
-                       .orElse(Collections.emptyMap());
```

Comment:
I wouldn't do this, as it means the execution of an additional lambda, whereas the `Collections.emptyMap()` method just returns the static `java.util.Collections#EMPTY_MAP` field (so there is no allocation cost saving which would otherwise justify the `orElse`).

Reply:
@0x006EA1E5 
Thank you for your suggestion.

But I have different understanding:
1. `lambda` is just a pointer to a method, it will not need extra cost.
2. In `orElse(doSomething())`, `doSomething()` will always be executed, it's a kind of waste (a method call will cost `Stack Frame`). Here is code to verify this:
```
public class Test {

    public static void main(String[] args) {
        System.out.println("Start orElse");
        String orElseResult = Optional.of("string")
                .orElse(doSomething());
        System.out.println("orElseResult = " + orElseResult);

        System.out.println();
        System.out.println("Start orElseGet");
        String orElseGetResult = Optional.of("string")
                .orElseGet(Test::doSomething);
        System.out.println("orElseGetResult = " + orElseGetResult);
    }
    public static String doSomething() {
        System.out.println("doSomething");
        return "doSomething";
    }
}
```
Execution result:
![image](https://user-images.githubusercontent.com/13167207/195018736-84a31e2a-c912-4319-90bb-b062d5461abc.png)

Reply:
Got it. Thank you for your detailed explanation.